spectator_mode/init.lua

```diff
@@ -122,13 +122,18 @@ minetest.register_on_leaveplayer(function(player)
 	end
 end)
 
-minetest.register_on_joinplayer(function(player)
-	if not minetest.check_player_privs(player:get_player_name(), { spectate = true }) then
-		return
+local function join_player(player)
+	if minetest.check_player_privs(player:get_player_name(), { spectate = true }) then
+		hide_player(player)
+		irc.send_join_part = false
```

Comment:
We probably shouldn't assume the default value is true, which would mean storing it in a local variable to restore later